lib/graphql/execution/lazy/lazy_method_map.rb

```diff
@@ -6,7 +6,14 @@ class Lazy
       # The result of this lookup is cached for future resolutions.
       class LazyMethodMap
         def initialize
-          @storage = {}
+          @storage = Hash.new do |h, value_class|
+            registered_superclass = h.each_key.find { |lazy_class| value_class < lazy_class }
+            if registered_superclass.nil?
+              h[value_class] = nil
+            else
+              h[value_class] = h[registered_superclass]
+            end
```

Comment:
If you want to measure the before/after, I'd be happy to consider this patch in another PR!